drivers/regulator/regulator_npm6001.c

```diff
@@ -221,7 +221,7 @@ static int regulator_npm6001_ldo0_set_voltage(const struct device *dev, int32_t
 		}
 	}
 
-	if (i == ARRAY_SIZE(ldo0_voltages)) {
+	if (val == 0xff) {
```

Comment:
why is this needed?

Reply:
I have tried to explain in the commit message and hence the PR description

Reply:
Sorry, but how is this change related to uninitialized variable? This is changing exit logic, which should be fine as is.

Reply:
`val == 0fxff` is more explicit than `i == ARRAY_SIZE(ldo0_votages)`  and hence the compiler does not complain. Without this change, the compiler throws the following error.

/home/balaji/oob-firmware/zephyr/drivers/regulator/regulator_npm6001.c:228:16: warning: 'val' may be used uninitialized [-Wmaybe-uninitialized]
  228 |         return i2c_reg_write_byte_dt(&config->i2c, NPM6001_LDO0VOUT, val);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/balaji/oob-firmware/zephyr/drivers/regulator/regulator_npm6001.c:214:17: note: 'val' was declared here
  214 |         uint8_t val;

Reply:
This error doesn't relate to val == 0xff, just the definition of val.
Do you get a warning if you just initialise val to zero?

Reply:
I don't get the warning if I initialize it to 0. It just needs some init value. I believe that 0xff is the correct (invalid) init value for this variable.

Reply:
if initialized to zero (instead of 0xff, which initially looks like a magic value) will resolve the issue.

Reply:
If you initialise val to zero, and leave line 224 unchanged, do you still get the warning?

Reply:
No. I have now pushed the change where I only initialize the `val` to `0`. Left line 224 unchanged.